docs: add details about HPA via KEDA (#9202)
rook/rook: Documentation/ceph-monitoring.md

    prometheus: k8s
[...]

> ```

Reviewer: It would be nice to have a table of contents at the top so new topics like this can be more easily discovered, but that is really a separate question from this PR.

> ### Horizontal Pod Scaling using Kubernetes Event-driven Autoscaling aka KEDA

Reviewer: ```suggestion
### Horizontal Pod Scaling using Kubernetes Event-driven Autoscaling (KEDA)
```

> Using metrics exported from the Prometheus service, the horizontal pod scaling can use the custom metrics other than CPU and memory consumption. It can be done with help of Promethues Scaler provided by the [KEDA](https://keda.sh/docs/1.4/scalers/prometheus/). The KEDA deployed via helm chart, refer [here](https://keda.sh/docs/2.4/deploy/) for details.

Reviewer: These version links will get outdated as they have more releases. Is there a "latest" or similar link we can use?
```suggestion
Using metrics exported from the Prometheus service, the horizontal pod scaling can use the custom metrics other than CPU and memory consumption. It can be done with help of Prometheus Scaler provided by [KEDA](https://keda.sh/docs/2.4/scalers/prometheus/). See the [KEDA deployment guide](https://keda.sh/docs/2.4/deploy/) for details.
```

> Following is sample for `ScaledObject`for autoscaling RGW:

Reviewer: ```suggestion
Following is an example to autoscale RGW:
```

> ```YAML

Reviewer: Please also create this yaml in the [monitoring examples folder](https://github.com/rook/rook/tree/master/cluster/examples/kubernetes/ceph/monitoring).
